### PR TITLE
Rest API endpoints with hawk authentication

### PR DESCRIPTION
**Pull request by ambrozic**
## Description of change

- added Rest API with hawk authentication
- added rest api endpoint for enquiries data

## Test instructions

_What should I see? (If visible changes)_

### CODE REVIEW

File: app/enquiries/tests/test_auth.py [hunk 1]
```diff
@@ -0,0 +1,461 @@
...
+        response = api_client.get(
+            _url(),
+            content_type="",
+            HTTP_AUTHORIZATION=sender.request_header,
+        )
+
+        assert response.status_code == status.HTTP_200_OK
+        assert response.json() == {"content": "hawk-test-view-without-scope"}
+
+    def test_empty_object_returned_with_authentication(self, api_client, settings):
```
Comment (mforner13) [inline app/enquiries/tests/test_auth.py:300]: In these two tests it looks like you're making sure an object which is **not** empty is returned, so the names seem a bit misleading. Apologies if I've misunderstood.

File: app/enquiries/tests/test_views.py [hunk 2]
```diff
@@ -717,3 +713,129 @@ def test_import_template(self):
...
+@pytest.mark.django_db
+class TestAPIEnquiries:
+    def test_enquiries_list_unauthorized(self, api_client):
+        """Should return 401"""
+        url = reverse("api-v1-enquiries")
+        response = api_client.get(url)
+        assert response.status_code == status.HTTP_401_UNAUTHORIZED
+
+    def test_enquiries_list_forbidden(self, api_client, settings):
+        """Should return 200"""
```
Comment (mforner13) [inline app/enquiries/tests/test_views.py:749]: Looks like it should return a 403?

**Review by mforner13 — COMMENTED** [review]
A few comments, + adding adding the hawk credential env vars to `sample.env` :)

File: app/enquiries/views.py [hunk 3]
```diff
@@ -93,6 +97,13 @@ def get_enquiry_field(name):
     return {"name": name, "choices": filter_config[name].choices}
 
 
+class Pagination(PageNumberPagination):
+    """
+    Control API views pagination.
+    """
+    page_size_query_param = settings.ENQUIRIES_PAGE_SIZE_PARAM
```
[the remark this answers is not in the document]
Comment (ambrozic) [reply app/enquiries/views.py:104]: i want to control page size per request ie `enquiries?page_size=123` and i could not find a better way to do it. ideally this should be a setting in `settings.REST_FRAMEWORK`

File: app/settings/common.py [hunk 4]
```diff
@@ -227,6 +224,7 @@
 # App specific settings
 CHAR_FIELD_MAX_LENGTH = 255
 ENQUIRIES_PER_PAGE = env.int('ENQUIRIES_PER_PAGE', default=10)
+ENQUIRIES_PAGE_SIZE_PARAM = env.int('ENQUIRIES_PAGE_SIZE_PARAM', default='page_size')
```
[the remark this answers is not in the document]
Comment (ambrozic) [reply app/settings/common.py:227]: same as above, i wanted to define a query parameter name that controls page size on request.

File: app/enquiries/tests/test_auth.py (hunk 1 above)
Comment (ambrozic) [reply app/enquiries/tests/test_auth.py]: thank you. renamed. was copy pasting tests from DH API :)

**Review by mforner13 — APPROVED** [review]
👍

**Review by elcct — APPROVED** [review]

**Pull request merged**